### PR TITLE
added recovery status to table info

### PR DESCRIPTION
**Pull request by mikethebeer**
screen shot 2015-11-27 at 11 14 35

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -5,6 +5,8 @@ Changes for Crate Admin Interface
 Unreleased
 ==========
 
+ - display current shard recovery status on table info page
```
Comment (celaus) [inline CHANGES.txt]: :D this could be more enthusiastic (and descriptive maybe)

File: app/scripts/services/tableinfo.js [hunk 2]
```diff
@@ -104,7 +104,37 @@ angular.module('tableinfo', ['sql'])
         if (numUnassignedShards > 0 || numUnderreplicatedShards > 0) return 'warning';
         return 'good';
       }());
-
+    
+      var recovery_percent = (function() {
+        if (recovery === null || typeof recovery === 'undefined') return;
```
Comment (celaus) [inline app/scripts/services/tableinfo.js]: if recovery is a list, you can also use if(!recovery)

File: app/scripts/services/tableinfo.js [hunk 3]
```diff
@@ -104,7 +104,37 @@ angular.module('tableinfo', ['sql'])
...
+          return data.recovery_stage === 'INIT';
+        });
+        var count_shards = init_filter.length;
+        
+        var recovered_filter = recovery.filter(function(data, idx) {
+          return data.recovery_stage !== 'INIT' && data.recovery_stage !== 'DONE';
+        });
+        count_shards += recovered_filter.length;
+        var recovered = recovered_filter.reduce(function(memo, data, idx) {
+          return memo + (data.recovery_percent / recovered_filter.length);
```
Comment (celaus) [inline app/scripts/services/tableinfo.js]: Division by 0 possible if all shards are in done or in init?

File: app/scripts/services/tableinfo.js [hunk 4]
```diff
@@ -104,7 +104,37 @@ angular.module('tableinfo', ['sql'])
         if (numUnassignedShards > 0 || numUnderreplicatedShards > 0) return 'warning';
         return 'good';
       }());
-
+    
+      var recovery_percent = (function() {
+        if (recovery === null || typeof recovery === 'undefined') return;
+        var init_filter = recovery.filter(function(data, idx) {
+          return data.recovery_stage === 'INIT';
```
Comment (celaus) [inline app/scripts/services/tableinfo.js]: case sensitive check! change to insensitive (SQL or here).

File: app/scripts/services/tableinfo.js [hunk 5]
```diff
@@ -104,7 +104,37 @@ angular.module('tableinfo', ['sql'])
...
+        });
+        var count_shards = init_filter.length;
+        
+        var recovered_filter = recovery.filter(function(data, idx) {
+          return data.recovery_stage !== 'INIT' && data.recovery_stage !== 'DONE';
```
Comment (celaus) [inline app/scripts/services/tableinfo.js]: case sensitivity

File: app/scripts/services/tableinfo.js [hunk 6]
```diff
@@ -104,7 +104,37 @@ angular.module('tableinfo', ['sql'])
...
+        });
+        count_shards += recovered_filter.length;
+        var recovered = recovered_filter.reduce(function(memo, data, idx) {
+          return memo + (data.recovery_percent / recovered_filter.length);
+        }, 0);
+        
+        var done_filter = recovery.filter(function(data, idx) {
+          return data.recovery_stage === 'DONE';
```
Comment (celaus) [inline app/scripts/services/tableinfo.js]: case sensitivity

File: app/scripts/services/tableinfo.js (hunk 3 above)
Comment (mikethebeer) [inline app/scripts/services/tableinfo.js]: reduce is just an iterator. this line doesn't get executed if contains 0 elements

File: app/scripts/services/tableinfo.js (hunk 6 above)
Comment (mikethebeer) [inline app/scripts/services/tableinfo.js]: shouldn't be a problem because it won't change

File: app/scripts/services/tableinfo.js (hunk 3 above)
Comment (celaus) [inline app/scripts/services/tableinfo.js]: True, did not see that this is the same collection :+1:

File: app/scripts/services/tableinfo.js (hunk 6 above)
Comment (celaus) [inline app/scripts/services/tableinfo.js]: Famous last words! :hear_no_evil:

File: app/views/tables.html [hunk 7]
```diff
@@ -112,6 +112,16 @@
           </div>
         </div>
       </div><!-- /.row -->
+      <div class="row">
+        <div class="col-sm-12">
+          <div class="panel panel-info text-right">
+            <div class="panel-heading">
+              <p class="announcement-heading inline-block" data-toggle="tooltip" rel="tooltip" data-original-title="{{table.recovery_percent | number:0}}">{{table.recovery_percent | number:0}}</p>
+              <p class="announcement-text">% recovery</p>
```
Comment (chaudum) [inline app/views/tables.html]: % recovery done

**Comment by celaus** [comment]
ack :+1:  squash and merge! :)

**Pull request merged**